gc/base/HeapLinkedFreeHeader.hpp

```diff
@@ -178,47 +157,25 @@ class MM_HeapLinkedFreeHeader
 	 */
 	MMINLINE MM_HeapLinkedFreeHeader* afterEnd() { return (MM_HeapLinkedFreeHeader*)( ((uintptr_t)this) + getSize() ); }
 
-	/**
-	 * Mark the specified region of memory as all single slot holes
-	 * @param[in] addrBase the address to start marking from
-	 * @param freeEntrySize the number of bytes to be consumed
-	 */
-	MMINLINE static void
-	fillWithSingleSlotHoles(void* addrBase, uintptr_t freeEntrySize)
-	{
-#if defined(SPLIT_NEXT_POINTER)
-		uint32_t *freeSlot = (uint32_t *) addrBase;
-		while(freeEntrySize) {
-			*freeSlot++ = J9_GC_SINGLE_SLOT_HOLE;
-			freeEntrySize -= sizeof(uint32_t);
-		}
-#else /* defined(SPLIT_NEXT_POINTER) */
-		uintptr_t *freeSlot = (uintptr_t*) addrBase;
-		while(freeEntrySize) {
-			*freeSlot++ = J9_GC_SINGLE_SLOT_HOLE;
-			freeEntrySize -= sizeof(uintptr_t);
-		}
-#endif /* defined(SPLIT_NEXT_POINTER) */
-	}
-
 	/**
 	 * Mark the specified region of memory as walkable dark matter
 	 * @param[in] addrBase the address where the hole begins
 	 * @param[in] freeEntrySize the number of bytes to be consumed (must be a multiple of sizeof(uintptr_t))
 	 * @return The header written or null if the space was too small and was filled with single-slot holes
 	 */
 	MMINLINE static MM_HeapLinkedFreeHeader*
-	fillWithHoles(void* addrBase, uintptr_t freeEntrySize)
+	fillWithHoles(void* addrBase, uintptr_t freeEntrySize, bool compressed)
 	{
-		MM_HeapLinkedFreeHeader *freeEntry = NULL;
+		MM_HeapLinkedFreeHeader *freeEntry = (MM_HeapLinkedFreeHeader *)addrBase;
 		if (freeEntrySize < sizeof(MM_HeapLinkedFreeHeader)) {
-			/* Entry will be abandoned. Recycle the remainder as single slot entries */
-			fillWithSingleSlotHoles(addrBase, freeEntrySize);
+			if (0 != freeEntrySize) {
```

Comment:
Can we have `while` instead of `if` here ? we might increase size for `MM_HeapLinkedFreeHeader` for experimental purpose

Reply:
Will do - I'll change the assert to make sure it's uintptr_t aligned rather than exactly uintptr_t.

Reply:
Done.